x/farming/spec/03_state_transitions.md

```diff
@@ -22,55 +22,39 @@ const (
-  - `StartEpochId` is modified to the `EpochId` of the current epoch
-  - unstake executed immediately and `StakedCoins` are reduced accordingly
-- Manual reward withdrawal : When a farmer request a reward withdrawal
-  - accumulated rewards until last epoch are immediately withdrawn
-  - `StartEpochId` is modified to the `EpochId` of the current epoch
-
-## Accumulated Reward Calculation
-
-- Accumulated Unit Reward : AUR represents accumulated rewards(for each staking coin) of a staking position with amount 1.
-- AUR for each staking coin for each epoch can be calculated as below
-  - ![](https://latex.codecogs.com/svg.latex?\Large&space;\sum_{i=0}^{now}\frac{TR_i}{TS_i})
-    - ![](https://latex.codecogs.com/svg.latex?\Large&space;i) : each `EpochId`
-    - ![](https://latex.codecogs.com/svg.latex?\Large&space;now) : current `EpochId`
-    - ![](https://latex.codecogs.com/svg.latex?\Large&space;TS_i) : total staking amount of the staking coin for epoch i
-    - ![](https://latex.codecogs.com/svg.latex?\Large&space;TR_i) : total reward amount of the staking coin for epoch i
-- Accumulated rewards from any staking position can be calculated from AUR and the staking amount of the position as below
-  - ![](https://latex.codecogs.com/svg.latex?\Large&space;x*\(\sum_{i=0}^{now}\frac{TR_i}{TS_i}-\sum_{i=0}^{start}\frac{TR_i}{TS_i}\))
-    - assuming constant staking amount for the staking epochs
-    - ![](https://latex.codecogs.com/svg.latex?\Large&space;x) : staking amount for the staking period
\ No newline at end of file
+## Stake
+
+When a farmer stakes an amount of coins, the following state transitions occur:
+
+- it reserves the amount of coins to the staking reserve pool account `StakingReservePoolAcc` 
+- it creates `QueuedStaking` object and stores the staking coins in `QueueStaking`, which are waiting in a queue until the end of epoch to move to `Staking` object
+- it imposes more gas if the farmer already has `Staking` with the same coin denom(see [07_params.md](07_params.md#DelayedStakingGasFee) for details)
+
+## Unstake
+
+When a farmer unstakes an amount of coins, the following state transitions occur:
+
+- it adds `Staking` and `QueueStaking` amounts to see if the unstaking amount is sufficient
+- it automatically withdraws rewards for the coin denom which are accumulated over the last epochs
+- it subtracts the unstaking amount of coins from `QueueStaking` first and if it is not sufficient then it subtracts from `Staking`
+- it releases the unstaking amount of coins to the farmer
+
+## Harvest (Reward Withdrawal)
+
+- it calculates `CumulativeUnitRewards` in `HistoricalRewards` object in order to get the rewards for the staking coin denom which are accumulated over the last epochs for the farmer
+- it releases the accumulated rewards to the farmer if it is not zero and decreases the `OutstandingRewards`
+- it sets `StartingEpoch` in `Staking` object
+
+## Reward Allocation
```

Comment:
I think it'd be good to add this sentence:
> If the sum of total calculated `EpochAmount`(or `EpochRatio` multiplied by the farming pool's balance) exceeds the farming pool's balance, then the reward allocation is skipped for that epoch.

Reply:
Let me add the line in the beginning of "Reward Allocation" section.